Update documentation with GA4GH

slowkow opened the pull request:
- Add GA4GH build steps to building.html
- Add new page called tools-that-use-picard.html

nh13 commented on tools-that-use-picard.html:
```diff
@@ -0,0 +1,28 @@
+---
```
Since this *actually* part of Picard, I would not include.

nh13 commented on index.html:
```diff
@@ -106,6 +106,7 @@
 				<li><a href="http://samtools.sourceforge.net/SAM1.pdf">SAM Format Specification</a></li>
 				<li><a href="http://sourceforge.net/mail/?group_id=246254">SAMTools mailing Lists</a></li>
 				<li><a href="http://samtools.sourceforge.net/">SAMTools Home Page</a></li>
+				<li><a href="tools-that-use-picard.html">Tools that use Picard</a></li>
```
Remove.

vdauwera replied:
No offense to the author, but I agree that we should not be in the business of compiling lists of external tools that use or wrap Picard -- that's a user support minefield.

vdauwera commented:
The part of the doc that explains how to get help also needs to be updated -- it still refers to the samtools mailing list in two places, but that's out of date -- people should ask for help on the GATK forum now.

vdauwera commented:
Done reviewing, back to you @slowkow

slowkow commented:
Hi Nils and Geraldine,

Thanks for your comments, I appreciate them.

I thought it would be useful to maintain a list of tools built on top of Picard, but perhaps this is not the appropriate place for such a list. Right now I know of [picardmetrics] and [RNA-SeQC]. Please let me know if you know any others. I might start a list elsewhere.

[picardmetrics]: https://github.com/slowkow/picardmetrics
[RNA-SeQC]: http://www.broadinstitute.org/cancer/cga/rna-seqc

nh13 commented:
Thanks @slowkow for the changes.  Would you like to squash your commits and then I will merge?

slowkow commented:
I'd be happy to squash them, but the repo seems to have some funny files that git thinks are modified and cannot be removed. I have not touched these files, but they are labeled "modified". This prevents me from doing the `git rebase` command needed to squash the commits.

I don't know why this is happening, but it prevents me from squashing the commits. Sorry about that!

The following tips do not solve the issue:
- https://stackoverflow.com/questions/11383094/unstaged-changes-left-after-git-reset-hard
- https://stackoverflow.com/questions/3807903/git-checkout-file-casing-in-file-name

## Stubborn files

```
git rebase -i
Cannot rebase: You have unstaged changes.
Please commit or stash them.
```

Here's the situation:

```
git status

On branch slowkow-gh-pages-patch-1
Your branch is up-to-date with 'origin/slowkow-gh-pages-patch-1'.
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git checkout -- <file>..." to discard changes in working directory)

	modified:   javadoc/picard/picard/analysis/CollectRawWGSMetrics.RawWgsMetrics.html
	modified:   javadoc/picard/picard/analysis/CollectRawWGSMetrics.html
	modified:   javadoc/picard/picard/analysis/class-use/CollectRawWGSMetrics.RawWgsMetrics.html
	modified:   javadoc/picard/picard/analysis/class-use/CollectRawWGSMetrics.html

no changes added to commit (use "git add" and/or "git commit -a")
```

## Cannot remove stubborn files

```
git checkout -- .
git status

On branch slowkow-gh-pages-patch-1
Your branch is up-to-date with 'origin/slowkow-gh-pages-patch-1'.
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git checkout -- <file>..." to discard changes in working directory)

	modified:   javadoc/picard/picard/analysis/CollectRawWGSMetrics.RawWgsMetrics.html
	modified:   javadoc/picard/picard/analysis/CollectRawWGSMetrics.html
	modified:   javadoc/picard/picard/analysis/class-use/CollectRawWGSMetrics.RawWgsMetrics.html
	modified:   javadoc/picard/picard/analysis/class-use/CollectRawWGSMetrics.html

no changes added to commit (use "git add" and/or "git commit -a")
```

Also fails:

```
git stash
Saved working directory and index state WIP on slowkow-gh-pages-patch-1: b6f613a update links
HEAD is now at b6f613a update links

git stash drop
Dropped refs/stash@{0} (cda76d1b06a7c46819fba0e1094b2d8cfc0d4b5c)

git status
On branch slowkow-gh-pages-patch-1
Your branch is up-to-date with 'origin/slowkow-gh-pages-patch-1'.
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git checkout -- <file>..." to discard changes in working directory)

	modified:   javadoc/picard/picard/analysis/CollectRawWGSMetrics.RawWgsMetrics.html
	modified:   javadoc/picard/picard/analysis/CollectRawWgsMetrics.html
	modified:   javadoc/picard/picard/analysis/class-use/CollectRawWgsMetrics.RawWgsMetrics.html
	modified:   javadoc/picard/picard/analysis/class-use/CollectRawWgsMetrics.html

no changes added to commit (use "git add" and/or "git commit -a")
```

## The filenames seem to be case-insensitive?

Interestingly, this command does not get rid of the file, but it does change the case of some of the characters in the filename:

```
git checkout -- javadoc/picard/picard/analysis/CollectRawWGSMetrics.RawWgsMetrics.html
git status

On branch slowkow-gh-pages-patch-1
Your branch is up-to-date with 'origin/slowkow-gh-pages-patch-1'.
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git checkout -- <file>..." to discard changes in working directory)

	modified:   javadoc/picard/picard/analysis/CollectRawWGSMetrics.html
	modified:   javadoc/picard/picard/analysis/CollectRawWgsMetrics.RawWgsMetrics.html
	modified:   javadoc/picard/picard/analysis/class-use/CollectRawWGSMetrics.RawWgsMetrics.html
	modified:   javadoc/picard/picard/analysis/class-use/CollectRawWGSMetrics.html

no changes added to commit (use "git add" and/or "git commit -a")
```

Notice that `CollectRawWGSMetrics.RawWgsMetrics.html` is now called `CollectRawWgsMetrics.RawWgsMetrics.html`. The **WGS** changed to **Wgs**.

jmarshall commented:
The `gh-pages` branch contains files [spelt both ways](https://github.com/broadinstitute/picard/tree/gh-pages/javadoc/picard/picard/analysis).  You are on a Mac, the files are colliding in your working tree, and you are out of luck.

Apparently the canonical spelling is **Wgs** (ec05212a4db28c5ec6824b9021790d6865ebc272).  So somebody needs to go and nuke the four five-month-old `CollectRawWGSMetrics*` legacy files from the **gh-pages** branch.  This will be easy on Unix or the GitHub web UI, and either moderately easy-ish or awful (requiring use of non-porcelain) on a Mac.

nh13 commented:
Fixed in this (https://github.com/broadinstitute/picard/pull/239) pull request, thank-you all for the investigation.
